### DISCUSSION TITLE
HyperSPI on STM32 or Raspberry Pi Pico

### DISCUSSION THREAD

**Discussion by newcgen** [awawa-dev/HyperSPI#3] [Ideas]
Hi!
It would be interesting to have an implementation
HyperSPI on other platforms?

**Comment by awawa-dev** [comment]
Hi
First I need to get a hand on such device ;) Generic awa-SPI protocol in HyperHDR is already implemented (third option beside custom esp32 and esp8266), just need a client. But I'm not sure if neopixelbus library supports STM32. And that could be a problem.

**Comment by newcgen** [comment]
Yes, neopixelbus does not support STM32. This is a not big deal.

**Comment by awawa-dev** [comment]
Unfortunately neopixelbus library is used for this project to drive the LED strip. I'm not saying it's impossible, I'm not familiar with that platform, maybe there is alternative that handles all a bunch of ws281x/sk6812 LED strips (just like for old-generation arduinos adalight sketches), but it would be required to rewrite some parts of code and then some manual testing. Just as it was done with the HyperSPI using Makuna library for esp32/esp8266. I can't imagine to implement it blindly without a device. Are there any STM32 advantages over esp8266, for example much lower price or 5v GPIO?

**Comment by newcgen** [comment]
The price of a black pill with STM32F401CCU6 is 3$. They hava USB. Some GPIO is 5v tolerant (only for input). But GPIO can be configured as open-drain output.
